onnx/defs/tensor/defs.cc

```diff
@@ -2538,164 +2538,178 @@ ONNX_OPERATOR_SET_SCHEMA(
-                          step_d_then = Div (two, D_float)
-                          step_h_then = Div (two, H_float)
-                          step_w_then = Div (two, W_float)
-
-                          step_d_half = Div (step_d_then, two)
-                          start_d_then = Add (minus_one, step_d_half)
-
-                          step_h_half = Div (step_h_then, two)
-                          start_h_then = Add (minus_one, step_h_half)
-
-                          step_w_half = Div (step_w_then, two)
-                          start_w_then = Add (minus_one, step_w_half)
+                  size_NCDHW_else = Concat <axis=0> (N, C, D, H, W)
+              }
+          >
+          N, C, D, H, W = Split <num_outputs: int=5> (size_NCDHW)
+
+          theta_3d = If (condition_is_2d) <
+              then_branch = g3 () => (theta_then) { # theta: N by 2 by 3 => N by 3 by 4
+                  # use of thetaN23 is a way to make shape inference happy when theta is N by 3 by 4.
+                  gather_idx_6 = Constant <value_ints: ints = [0, 1, 2, 0, 1, 2]> ()
+                  shape_23 = Constant <value_ints: ints = [2, 3]> ()
+                  gather_idx_23 = Reshape (gather_idx_6, shape_23)
+                  shape_N23 = Concat <axis=0>(N, shape_23)
+                  gather_idx_N23 = Expand (gather_idx_23, shape_N23)
+                  thetaN23 = GatherElements <axis=2> (theta, gather_idx_N23) # N by 2 by 3 => N by 3 by 2
+
+                  r1, r2 = Split <axis: int=1, num_outputs: int=2> (thetaN23) # N by 1 by 3
+                  r1_ = Squeeze (r1) # N by 3
+                  r2_ = Squeeze (r2)
+                  r11, r12, t1 = Split <axis: int=1, num_outputs: int=3> (r1_) # N by 1
+                  r21, r22, t2 = Split <axis: int=1, num_outputs: int=3> (r2_)
+
+                  r11_shape = Shape (r21)
+                  float_zero_1d_ = ConstantOfShape (r11_shape) # N by 1
+                  float_zero_1d = CastLike (float_zero_1d_, theta)
+                  float_one_1d = Add (float_zero_1d, one_f) # N by 1
+
+                  R1 = Concat <axis=1>(r11, r12, float_zero_1d, t1) # N by 4
+                  R2 = Concat <axis=1>(r21, r22, float_zero_1d, t2)
+                  R3 = Concat <axis=1>(float_zero_1d, float_zero_1d, float_one_1d, float_zero_1d)
+
+                  R1_ = Unsqueeze (R1, one_v) # N by 1 by 4
+                  R2_ = Unsqueeze (R2, one_v)
+                  R3_ = Unsqueeze (R3, one_v)
+                  theta_then = Concat <axis=1> (R1_, R2_, R3_) # N by 3 by 4
+                  # theta_then = Identity (theta)
+              },
+              else_branch = g4 () => (theta_else) {
+                  theta_else = Identity (theta)
+              }
+          >
+
+          int_two_1d = Constant <value_ints=[2]> ()
+          int_three_1d = Constant <value_ints=[3]> ()
+          int_five_1d = Constant <value_ints=[5]> ()
```

Comment:
Should these names be updated to follow the naming convention above?

Reply:
I find the `_1d` convention more understandable than the `_v` (not sure what it stands for).

Reply:
vector I guess? But I agree. Consistency would be great here

Reply:
good catch @justinchuby , they shall follow the naming rules set for this body. Agree with @gramalingam and changed to _1d

Reply:
code updated with this comment